src/types.rs

```diff
@@ -754,6 +760,24 @@ where
         }
     }
 
+    ///
+    /// Return a list of suitable keys, given a key id an signature algorithm
+    ///
+    pub fn filter_keys(&self, key_id: &Option<JsonWebKeyId>, signature_alg: &JS) -> Vec<&K> {
```

Comment:
I'd prefer not to add this to the stable public API. let's just make it `pub(crate)` for now.